src/main/java/com/synopsys/integration/detect/lifecycle/run/step/IntelligentModeStepRunner.java

```diff
@@ -125,10 +128,22 @@ public void runOnline(
 
         stepHelper.runToolIfIncluded(DetectTool.BINARY_SCAN, "Binary Scanner", () -> {
             BinaryScanStepRunner binaryScanStepRunner = new BinaryScanStepRunner(operationRunner);
-            Optional<String> scanId = binaryScanStepRunner.runBinaryScan(dockerTargetData, projectNameVersion, blackDuckRunData, binaryTargets);
 
-            if (scanId.isPresent()) {
-                scanIdsToWaitFor.add(scanId.get());
+            if (isMultipartUploadPossible(blackDuckRunData)) {
+                Optional<String> scanId = 
+                        binaryScanStepRunner.runBinaryScan(dockerTargetData, projectNameVersion, blackDuckRunData, binaryTargets);
+
+                if (scanId.isPresent()) {
+                    scanIdsToWaitFor.add(scanId.get());
+                }
+            } else {
+                Optional<CodeLocationCreationData<BinaryScanBatchOutput>> codeLocationData = 
+                        binaryScanStepRunner.runLegacyBinaryScan(dockerTargetData, projectNameVersion, blackDuckRunData, binaryTargets);
+
+                if (codeLocationData.isPresent()) {
+                    codeLocationAccumulator.addWaitableCodeLocations(codeLocationData.get());
+                    mustWaitAtBomSummaryLevel.set(true);
+                }
```

Comment:
What's the difference here between the **_runBinaryScan > get scan ID_** versus **_runLegacyBinaryScan > get writable code location_** ? If this question can be answered by reviewing the bigger, original binary upload MR (I didn't have a chance to review), please ignore as I intend to look at that in detail at a later time

Reply:
So the new library we are using returns us the scanID. This allows us to wait on the bom-status/{scanID} endpoint which gives us pretty much an instant answer on if the scan is done. The older blackduck-common library approach did not give us the scanID so we had to pull a bunch of notifications and search for if there was a notification that the scan completed. I think the code called it a codeLocation cause I think technically we were searching the notification list for the codeLocation name cause we didn't have the scanID.

Reply:
I see, thank you for that explanation!